homeassistant/components/homeassistant_hardware/silabs_multiprotocol_addon.py

```diff
@@ -55,15 +59,64 @@
 SAVE_DELAY = 10
 
 
-@singleton(DATA_ADDON_MANAGER)
-async def get_addon_manager(hass: HomeAssistant) -> MultiprotocolAddonManager:
+@singleton(DATA_MULTIPROTOCOL_ADDON_MANAGER)
+async def get_multiprotocol_addon_manager(
+    hass: HomeAssistant,
+) -> MultiprotocolAddonManager:
     """Get the add-on manager."""
     manager = MultiprotocolAddonManager(hass)
     await manager.async_setup()
     return manager
 
 
-class MultiprotocolAddonManager(AddonManager):
+class SyncAddonManager(AddonManager):
+    """Addon manager which supports synchronous operations for managing an addon."""
```

Comment:
We typically use synchronous / sync in a non-standard way in Home Assistant, where "sync" in Home Assistant jargon means something which blocks and hence should not be run from the asyncio event loop and "async" means something which either finishes in a very short time (a callback in Home Assistant jargon) or a coroutine function.

We may want to rename this class and its methods to not confuse it with the Home Assistant jargon, maybe `WaitingAddonManager` or something. Do you have any suggestions @MartinHjelmare ?

Reply:
Yeah, sounds good.